### DISCUSSION TITLE
Wrong queryString restoration for long numbers (is casting to int insteand maintain it as string)

### DISCUSSION THREAD

**Discussion by ProtocolNebula** [livewire/livewire#9058] [7. Bugs (archived)]
### Livewire version

v3.5.12 and 3.5.15

### Laravel version

v11.30.0

### Which PHP version are you using?

PHP 8.2

### Steps To Reproduce

1. Create a livewire component with the following:

```php
<?php
namespace App\Livewire;

use Livewire\Attributes\Validate;
use Livewire\Component;
use Illuminate\Http\Request;
use Illuminate\Support\Arr;

class Simple extends Component
{
    #[Validate('required|string|min:20|max:22')]
    public string $cups = '';

    protected $queryString = [
        'cups' => ['except' => ''],
    ];

    public function mount()
    {
        dd($this->cups);
    }
}
```

2. Mount it in a route, mine is "comparator":
```php
    Route::get('/comparator', function () {
        return view('comparator.index');
    })->name('comparator');
```

3. Try to access using the following url: `http://localhost:8000/comparator?cups=12345678901234567890`

OR: `https://4llwe.wirebox.app:8443/?cups=01234567890123456789` (unable to reproduce with the settings in the box)

## Hacky fix:

> Replace the previous mount method with:

```php
    public function mount(Request $request)
    {
        if ($request->has('cups')) {
            $this->cups = (string) $request->query('cups');
        }
        dd($this->cups);
    }
```

### Repository or Wirebox URL

https://wirebox.app/b/4llwe

### Additional information

Online example with the query: https://4llwe.wirebox.app:8443/?cups=01234567890123456789

The example does not work neither matches the versions

### Please confirm (incomplete submissions will not be addressed)

- [X] I have provided easy and step-by-step instructions to reproduce the bug.
- [X] I have provided code samples as text and NOT images.
- [X] I understand my bug report will be removed if I haven't met the criteria above.

**Comment by joshhanley** [comment]
@ProtocolNebula I'm not sure I understand what the problem is. Can you explain what is actually happening and what you are expecting to happen as well as some screenshots? Thanks!

**Reply by ProtocolNebula** [reply]
Hi @joshhanley, sure, I forget to explain the exact error.

When the `queryString` are restored to the corresponding properties, there is a casting from **string** to **integer** (but I'm on a `Component`, not a `Model`).

**Current behavior:**
![imagen](https://github.com/user-attachments/assets/c1d6842a-5f42-4489-ba17-63eeecff7d78)


**Expected behavior:**
![imagen](https://github.com/user-attachments/assets/3f093bc2-eccc-4217-98cc-0efa2dde9c8c)

---

**Hack:**

In the ticket I've put a solution, which is basically run this on the `mount` method:

```php
        if ($request->has('cups')) {
            $this->cups = (string) $request->query('cups');
        }
```

I hope this clarifies the ticket.
